### PR TITLE
Don’t select fallback avatars randomly

### PR DESCRIPTION
**Pull request by tillprochaska**
* Selects a fallback image from `public/avatars` if no URL is provided.
* The fallback image is based on the user’s ID (or any other key).

### CODE REVIEW

File: app/components/avatar/avatar.rb [hunk 1]
```diff
@@ -1,15 +1,33 @@
...
+      @key = key
     end
 
     private
 
-    attr_reader :url
+    attr_reader :key
+
+    def url
+      return fallback_url unless @url
```
Comment (roschaefer) [inline app/components/avatar/avatar.rb:24]:
```suggestion
      @url || fallback_url
```

File: app/components/avatar/avatar.rb [hunk 2]
```diff
@@ -1,15 +1,33 @@
...
-    include ComponentHelper
+  class Avatar < ApplicationComponent
+    FALLBACK_BASE_URL = '/avatars'
+    FALLBACK_IMAGES = [
+      'fallback-cat.jpg',
+      'fallback-dog.jpg',
+      'fallback-otter.jpg',
+      'fallback-seal.jpg',
+      'fallback-squirrel.jpg',
+    ]
```
Comment (roschaefer) [inline app/components/avatar/avatar.rb:13]:
```suggestion
    FALLBACK_IMAGES = {
      cat: 'fallback-cat.jpg',
      dog: 'fallback-dog.jpg',
      otter: 'fallback-otter.jpg',
      seal: 'fallback-seal.jpg',
      squirrel: 'fallback-squirrel.jpg',
    }


```

File: app/components/avatar/avatar.rb [hunk 3]
```diff
@@ -1,15 +1,33 @@
...
 
-    def initialize(url:)
+    def initialize(url: nil, key: 0)
```
Comment (roschaefer) [inline app/components/avatar/avatar.rb:14]:
```suggestion
    def initialize(url: nil, animal: nil)
```

File: app/components/avatar/avatar.rb [hunk 4]
```diff
@@ -1,15 +1,33 @@
...
-    attr_reader :url
+    attr_reader :key
+
+    def url
+      return fallback_url unless @url
+      @url
+    end
+
+    def fallback_url
+      file = FALLBACK_IMAGES[key % FALLBACK_IMAGES.length]
```
Comment (roschaefer) [inline app/components/avatar/avatar.rb:29]:
```suggestion
      file = FALLBACK_IMAGES[animal] || FALLBACK_IMAGES.values.sample
```

File: app/components/avatar/avatar.html.erb [hunk 5]
```diff
@@ -1,3 +1,3 @@
 <div class="Avatar">
-    <img src="<%= url || 'https://source.unsplash.com/random?cat' %>" />
+    <img src="<%= url %>" alt="" />
```
Comment (roschaefer) [inline app/components/avatar/avatar.html.erb:2]: Does it help in any way to have an empty `alt`?

File: app/components/avatar/avatar.rb (hunk 1 above)
Comment (drjakob) [reply app/components/avatar/avatar.rb:24]: Otter!

File: app/components/avatar/avatar.html.erb (hunk 5 above)
Comment (tillprochaska) [reply app/components/avatar/avatar.html.erb:2]: Yes, an image tag without an `alt` attribute is invalid HTML. An empty `alt` attribute indicates that the image is optional. This way, many screen readers will simply skip the element. If you completely omit the attribute, screen readers might announce there’s an image element, but aren’t able to provide any details what is shown – which is confusing and unnecessary.

File: app/components/avatar/avatar.rb (hunk 3 above)
Comment (tillprochaska) [reply app/components/avatar/avatar.rb:14]: Why would you want to specify a concrete fallback image?

**Pull request merged**